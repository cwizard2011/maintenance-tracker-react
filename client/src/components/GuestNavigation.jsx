import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const GuestNavigation = (props) => {
  const { showLoginModal, burgerToggle } = props;
  return (
    <nav className="nav">
      <div className="navWide">
        <div className="wideDiv">
          <Link to="/" title="Home page"><i className="fa icon-font">&#xf015;</i></Link>
          <a
            role="presentation"
            title="Login Users"
            onClick={showLoginModal}
            value="Show Login"
          >
          Login
          </a>
        </div>
      </div>
      <div className="navNarrow">
        <Link to="/" title="Home page"><i className="fa icon-font">&#xf015;</i></Link>
        <i className="fa fa-bars fa-2x icon-padding" role="presentation" onClick={burgerToggle} />
        <div className="narrowLinks">
          <a href="#" title="Login Users" onClick={showLoginModal} value="Show Login">Login</a>
        </div>
      </div>
    </nav>
  );
};

GuestNavigation.propTypes = {
  showLoginModal: PropTypes.func,
  burgerToggle: PropTypes.func,
};

export default GuestNavigation;
