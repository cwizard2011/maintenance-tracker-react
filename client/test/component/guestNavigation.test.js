import React from 'react';
import { shallow } from 'enzyme';
import GuestNavigation from '../../src/components/GuestNavigation';

describe('Guest Navigation', () => {
  it('render guest navigation correctly', () => {
    const wrapper = shallow(<GuestNavigation />);
    expect(wrapper).toMatchSnapshot();
  });
});
