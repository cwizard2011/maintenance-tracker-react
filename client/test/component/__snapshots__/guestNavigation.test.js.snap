// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guest Navigation render guest navigation correctly 1`] = `
<nav
  className="nav"
>
  <div
    className="navWide"
  >
    <div
      className="wideDiv"
    >
      <Link
        replace={false}
        title="Home page"
        to="/"
      >
        <i
          className="fa icon-font"
        >
          
        </i>
      </Link>
      <a
        role="presentation"
        title="Login Users"
        value="Show Login"
      >
        Login
      </a>
    </div>
  </div>
  <div
    className="navNarrow"
  >
    <Link
      replace={false}
      title="Home page"
      to="/"
    >
      <i
        className="fa icon-font"
      >
        
      </i>
    </Link>
    <i
      className="fa fa-bars fa-2x icon-padding"
      role="presentation"
    />
    <div
      className="narrowLinks"
    >
      <a
        href="#"
        title="Login Users"
        value="Show Login"
      >
        Login
      </a>
    </div>
  </div>
</nav>
`;
